### PR TITLE
TEST: integration test JackknifeAfterBootstrapRegressor

### PR DESCRIPTION
**Pull request by jawadhussein462**
# Description

Implement Mapie v1 JackknifeAfterBootstrapRegressor integration tests

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist

- [] I have read the [contributing guidelines](https://github.com/scikit-learn-contrib/MAPIE/blob/master/CONTRIBUTING.rst)
- [] I have updated the [HISTORY.rst](https://github.com/scikit-learn-contrib/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/scikit-learn-contrib/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully and without warnings : `make doc`

### CODE REVIEW

File: mapie_v1/integration_tests/tests/test_regression.py [hunk 1]
```diff
@@ -195,56 +184,148 @@ def test_intervals_and_predictions_exact_equality_cross(params_cross):
...
     else:
         raise ValueError(f"Unknown strategy key: {strategy_key}")
 
+    v0_params = filter_params(MapieRegressorV0.__init__, v0_params)
+    v0 = MapieRegressorV0(**v0_params)
+
     return v0, v1
+
+
+def evaluate_models(
```
Comment (Valentin-Laurent) [inline mapie_v1/integration_tests/tests/test_regression.py:280]: `evaluate_models` is a bit of a vague name

File: mapie_v1/integration_tests/tests/test_regression.py [hunk 2]
```diff
@@ -195,56 +184,148 @@ def test_intervals_and_predictions_exact_equality_cross(params_cross):
...
+]
 
-    v0_preds, v0_pred_intervals = v0.predict(X_cross, **v0_predict_params)
 
-    v1_pred_intervals = v1.predict_set(X_cross, **v1_predict_set_params)
-    if v1_pred_intervals.ndim == 2:
-        v1_pred_intervals = np.expand_dims(v1_pred_intervals, axis=2)
-    v1_preds = v1.predict(X_cross, **v1_predict_params)
+@pytest.mark.parametrize("params_jackknife", params_test_cases_jackknife)
+def test_jackknife_after_bootstrap_consistency(params_jackknife):
```
Comment (Valentin-Laurent) [inline mapie_v1/integration_tests/tests/test_regression.py:243]: We should have the same names for similar tests

**Review by Valentin-Laurent — COMMENTED** [review]
Good job 🤟🏻
Few remarks here and in the code:

- Run `make v1-type-check`
- Missing test cases:
  - Method = minmax (the most important missing one I think)
  - Conf score = absolute
  - "allow_infinite_bounds": True,

File: mapie_v1/integration_tests/tests/test_regression.py (hunk 1 above)
Comment (jawadhussein462) [reply mapie_v1/integration_tests/tests/test_regression.py]: replacer by compare_model_predictions_and_intervals

File: mapie_v1/integration_tests/tests/test_regression.py (hunk 2 above)
Comment (jawadhussein462) [reply mapie_v1/integration_tests/tests/test_regression.py]: replaced by test_intervals_and_predictions_exact_equality_jackknife

**Review by Valentin-Laurent — APPROVED** [review]

**Pull request merged**